compiler/src/compilers/cpp_compiler.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(unused)]
+
+//use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+use super::compiler::Compiler;
+use super::super::data::input_data::InputData;
+use super::super::data::output_data::OutputData;
+use super::super::storage::name_generator::generate_filename;
+
+pub struct CppCompiler {
+
+}
+
+impl Compiler for CppCompiler {
+    fn compile(&self, input_data: &InputData) -> OutputData {
+        
+        let mut output_data = OutputData {
+            status_code: Some(-1),
+            compiled_file_name: PathBuf::from(""),
+            stdout: String::new(),
+            stderr: String::new(),
+        };
+
+        let bin_file_name = generate_filename().to_owned();
+        
+        let mut full_bin_filename = PathBuf::new();
+        full_bin_filename.push(&input_data.compiled_directory_path);
+        full_bin_filename.push(&bin_file_name);
+
+        let compiler_output = Command::new("g++")
+                .args(&[&input_data.compiler_options, 
```

Comment:
There is a problem with compiler_options containing "".  
g++ takes "" as a file name.

Reply:
Thanks a lot! Fix this problem - add if statement